server.js

```diff
@@ -1,35 +1,215 @@
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
-mongoose.connect(mongoUrl);
-mongoose.Promise = Promise;
+import express from "express"
+import cors from "cors"
+import mongoose from "mongoose"
+import expressListEndpoints from "express-list-endpoints"
+import { Restaurant } from "./models/Restaurant"
+import michelinData from "./data/michelin-data.json"
+
+//Set up MongoURL and localhost
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/restaurants"
+mongoose.connect(mongoUrl)
+mongoose.Promise = Promise
+
+if (process.env.RESET_DB) {
+  const seedDatabase = async () => {
+    await Restaurant.deleteMany()
+    //insert each document in the array into the collection
+    await Restaurant.insertMany(michelinData)
+  }
+  seedDatabase()
+}
 
 // Defines the port the app will run on. Defaults to 8080, but can be overridden
 // when starting the server. Example command to overwrite PORT env variable value:
 // PORT=9000 npm start
-const port = process.env.PORT || 8080;
-const app = express();
+const port = process.env.PORT || 3000
+const app = express()
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors());
-app.use(express.json());
+app.use(cors())
+app.use(express.json())
 
-// Start defining your routes here
+// Endpoint listing and guidance for using query endpoints
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
-});
+  const endpoints = expressListEndpoints(app).map((endpoint) => {
+    if (endpoint.path === "/restaurants") {
+      endpoint.query = {
+        description:
+          "There are 100 restaurants displayed per page in this endpoint. To see another page, add query endpoint like this '?page=3'.",
+      }
+    }
+
+    if (endpoint.path === "/restaurants/:query") {
+      endpoint.query = {
+        description:
+          "Add query endpoint to search for restaurant name or ID. If name includes more than one word, add space or '%20' between the words. ID should be a number between 1 and 6,700",
+      }
+    }
+
+    if (endpoint.path === "/cuisines/:cuisine") {
+      endpoint.query = {
+        description:
+          "Add query endpoint to search for cuisine type. See endpoint '/cuisines' for examples of different cuisines. If cuisine type includes more than one word, add space or '%20' between the words.",
+      }
+    }
+
+    if (endpoint.path === "/locations/:location") {
+      endpoint.query = {
+        description:
+          "Add query endpoint to search for city or country. If location name includes more than one word, add space or '%20' between the words.",
+      }
+    }
+
+    return endpoint
+  })
+  res.json(endpoints)
+})
+
+// All restaurants grouped by 100 per page
+app.get("/restaurants", async (req, res) => {
+  //Parse query parameters for pagination
+  const page = parseInt(req.query.page) || 1 //Defaults to page 1 if invalid
+  const limit = 100 //Display 100 restaurants per page
```

Comment:
I like that you put a limit to how many restaurants are showing up. That will help the frontend display it more organized to the user than having way to many restaurants on one page. 🥣